.gitlab-ci.yml

```diff
@@ -50,13 +50,21 @@ stages:
     - yes | ~/android_sdk/cmdline-tools/latest/bin/sdkmanager --licenses || true
     - echo "no" | ~/android_sdk/cmdline-tools/latest/bin/avdmanager --verbose create avd --force --name "$EMULATOR_NAME" --package "$ANDROID_EMULATOR_IMAGE"
   run-instrumented-integration:
+    - set +e
+    - exit_code=0
     - $ANDROID_HOME/emulator/emulator -avd "$EMULATOR_NAME" -grpc-use-jwt -no-snapstorage -no-audio -no-window -no-boot-anim -verbose -qemu -machine virt &
-    - GRADLE_OPTS="-Xmx3072m" ./gradlew :instrumented:integration:assembleDebug :instrumented:integration:assembleDebugAndroidTest --stacktrace --no-daemon $( (( $ANDROID_API <= 23 )) && echo "-Puse-api21-java-backport -Puse-desugaring" )
+    - GRADLE_OPTS="-Xmx3072m" ./gradlew :instrumented:integration:assembleDebug :instrumented:integration:connectedDebugAndroidTest --stacktrace --no-daemon $( (( $ANDROID_API <= 23 )) && echo "-Puse-api21-java-backport -Puse-desugaring" ) || exit_code=$?
+    - $ANDROID_HOME/platform-tools/adb emu kill
+    - if [[ "$exit_code" -ne 0 ]]; then exit 1; fi
+    - exit 0
+  run-core-it-instrumented:
```

Comment:
I am confused by what we have (mainly because of the naming/split) and have a few questions:

* what `it` in the `core-it` means? Is it `integration`? If yes, then we kind of have `run-instrumented-integration` snippet and `run-core-it[integration]-instrumented` snippet. Both of them are for integration + instrumented. Maybe we should replace `it` with full form for the clarity?
* As a consequence from the point above: what is the difference between the two? If we need to add a test which is both instrumented + integration, where should it go?

Reply:
the `instrumented-integration` are the old integration tests which we can find in the `instrumented/integration` package. They use to be running in Bitrise. I think from now on any integration test will be added in the `reliability` folder in one of the modules there. If you are asking about the difference between `instrumented` and `integration` we should not confuse these terms. `integration` is any test that tests a feature in its whole (different than `unit test`) whereas `instrumented` tells about the nature of the runner that executes that test, in our case that being the AndroidJUnitRunner which executes the integration tests on the Emulator itself.

Reply:
regarding your comment I think I can re - name a bit the scripts there to make more sense.

Reply:
My question is not about `instrumented` vs `integration`, my question is solely about 2 different modules (+ related jobs) which are both `instrumented` + `integration`.

I would also suggest renaming `core-it` module to expand the `it` term used there.

Reply:
the `core-it` is more to be in sync with the previous `single-fit` (single feature integration test). Do you mean to have something like `core-it-instrumented` ?